src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -314,14 +354,15 @@ private GapicSurfaceTestCaseView createTestCaseView(
             .resourcesVarName(namer.localVarName(Name.from("resources")))
             .build());
 
-    if (methodContext.getFeatureConfig().useResourceNameFormatOption(resourcesField)) {
-      String resourceName = ResourceNameUtil.getResourceName(resourcesField);
-      Name resourceNameName = Name.upperCamel(resourceName);
+    if (methodContext.getFeatureConfig().useResourceNameFormatOption(resourcesFieldConfig)) {
+      String resourceClassName =
+          namer.getResourceTypeClassName(resourcesFieldConfig.getEntityName());
+      Name resourceName = namer.getResourceTypeName(resourcesFieldConfig.getEntityName());
       resourceTypeName =
           methodContext
               .getTypeTable()
               .getAndSaveNicknameForTypedResourceName(
-                  resourcesField, resourcesField.getType().makeOptional(), resourceName);
+                  resourcesField, resourcesField.getType().makeOptional(), resourceClassName);
```

Comment:
Kept getResourceTypeName which returns the Name object, removed getResourceTypeClassName and added a getAndSave method to Surfacenamer that accepts the entityName.

I think we need to keep getResourceTypeName returning the Name object, not the result of className(...), because after calling className() we don't know the casing.